code/go/0chain.net/smartcontract/storagesc/models.go

```diff
@@ -197,22 +199,23 @@ func (sc *StorageChallenge) Save(state cstate.StateContextI, scAddress string) e
 type ValidationNode struct {
 	ID                string             `json:"id"`
 	BaseURL           string             `json:"url"`
+	ProviderType      spenum.Provider    `json:"provider_type"`
```

Comment:
Validation node is always a validator, why do you need to add this field?

Reply:
The main point is that we are using the same MPT key for all providers. 
```go
func GetKey(id string) datastore.Key {
	return "provider:" + id
}
```

So say when the user tries to say add a validator with the same id as an already existing blobber, then the sc will get the blobber object and put it into a validator object. By the error message returned, we can tell whether `add_validator` should fail because there is already a validoator with the same id or a blobber with that id.

In that way, it is easy to ensure that all providers have different ids.

Later we might be only interested in looking at the Provider part but still want to know what we are looking at.